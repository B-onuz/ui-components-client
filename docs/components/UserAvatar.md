# UserAvatar (Component)

Componente de Avatar

## Como utilizar

```js
import UserAvatar from 'ui-components/lib/components/UserAvatar'
```

```jsx
import UserAvatar from './UserAvatar'
import { Box, Flex } from 'reflexbox'

export default {
  title: 'UserAvatar',
  component: UserAvatar,
}

export const UserAvatarExample = () => <UserAvatar userName="John" />
```

## Variações

### Color

```jsx
<UserAvatar color={color} />
```

### DisplayName

```jsx
<Tooltip displayName="name or title" />
```

Escala no tema: `theme.variantScale`

## Styled-system props

- space https://styled-system.com/table#space
