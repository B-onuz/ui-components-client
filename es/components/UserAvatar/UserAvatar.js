"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _variants = require("./variants");

var _Typography = _interopRequireDefault(require("../Typography"));

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/UserAvatar/UserAvatar.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  ", "\n  width: 40px;\n  padding: 10px 0;\n  text-align: center;\n  background: ", ";\n  border-radius: 50%;\n  margin-right: 10px;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  ", "\n  display: flex;\n  align-items: center;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledUserAvatar = _styledComponents["default"].div(_templateObject(), _styledSystem.space);

var InitialsWrapper = _styledComponents["default"].div(_templateObject2(), _variants.color, function (_ref) {
  var theme = _ref.theme,
      color = _ref.color;
  return "" + theme.colors[color];
});

var UserAvatar = function UserAvatar(_ref2) {
  var children = _ref2.children,
      userName = _ref2.userName,
      color = _ref2.color,
      displayName = _ref2.displayName,
      nameColor = _ref2.nameColor,
      rest = _objectWithoutPropertiesLoose(_ref2, ["children", "userName", "color", "displayName", "nameColor"]);

  var _useState = (0, _react.useState)(),
      nameInitials = _useState[0],
      setNameInitials = _useState[1];

  (0, _react.useEffect)(function () {
    handleInitials(userName);
  }, []);

  var handleInitials = function handleInitials(name) {
    var initials = name.match(/\b\w/g) || [];
    initials = initials.length > 1 ? ((initials.shift() || '') + (initials.pop() || '')).toUpperCase() : name.substring(0, 2).toUpperCase();
    setNameInitials(initials);
  };

  return /*#__PURE__*/_react["default"].createElement(StyledUserAvatar, _extends({}, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 37,
      columnNumber: 5
    }
  }), /*#__PURE__*/_react["default"].createElement(InitialsWrapper, {
    color: color,
    m: 0,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 38,
      columnNumber: 7
    }
  }, nameInitials), !!displayName && /*#__PURE__*/_react["default"].createElement(_Typography["default"], {
    color: nameColor,
    m: 0,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 42,
      columnNumber: 9
    }
  }, displayName));
};

UserAvatar.defaultProps = {
  color: 'black',
  nameColor: 'black'
};
var _default = UserAvatar;
exports["default"] = _default;