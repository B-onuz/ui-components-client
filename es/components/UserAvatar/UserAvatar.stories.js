'use strict'

Object.defineProperty(exports, '__esModule', {
  value: true,
})
exports.UserAvatarWithName = exports.UserAvatarExample = exports['default'] = void 0

var _react = _interopRequireDefault(require('react'))

var _UserAvatar = _interopRequireDefault(require('./UserAvatar'))

var _reflexbox = require('reflexbox')

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/UserAvatar/UserAvatar.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

var _default = {
  title: 'UserAvatar',
  component: _UserAvatar['default'],
}
exports['default'] = _default

var UserAvatarExample = function UserAvatarExample() {
  return /*#__PURE__*/ _react['default'].createElement(_UserAvatar['default'], {
    userName: 'John',
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 10,
      columnNumber: 40,
    },
  })
}

exports.UserAvatarExample = UserAvatarExample

var UserAvatarWithName = function UserAvatarWithName() {
  return /*#__PURE__*/ _react['default'].createElement(
    _reflexbox.Flex,
    {
      flexDirection: 'column',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 13,
        columnNumber: 3,
      },
    },
    /*#__PURE__*/ _react['default'].createElement(_UserAvatar['default'], {
      userName: 'John The Doe',
      color: 'primary',
      displayName: 'Client',
      m: 2,
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 14,
        columnNumber: 5,
      },
    }),
    /*#__PURE__*/ _react['default'].createElement(_UserAvatar['default'], {
      userName: 'John Doe The',
      displayName: 'Lawyer',
      nameColor: 'primary',
      m: 2,
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 15,
        columnNumber: 5,
      },
    }),
    /*#__PURE__*/ _react['default'].createElement(_UserAvatar['default'], {
      userName: 'Example Lawyer',
      displayName: 'Lawyer',
      m: 2,
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 16,
        columnNumber: 5,
      },
    }),
    /*#__PURE__*/ _react['default'].createElement(_UserAvatar['default'], {
      userName: 'Client',
      displayName: 'Client',
      color: 'primary',
      nameColor: 'primary',
      m: 2,
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 17,
        columnNumber: 5,
      },
    })
  )
}

exports.UserAvatarWithName = UserAvatarWithName
