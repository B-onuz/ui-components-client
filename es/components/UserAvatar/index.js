"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _UserAvatar = _interopRequireDefault(require("./UserAvatar"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _UserAvatar["default"];
exports["default"] = _default;