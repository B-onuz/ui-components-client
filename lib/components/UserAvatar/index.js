"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _UserAvatar = _interopRequireDefault(require("./UserAvatar"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _UserAvatar["default"];
exports["default"] = _default;
module.exports = exports.default;