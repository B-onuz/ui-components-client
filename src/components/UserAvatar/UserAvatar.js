import React, { useState, useEffect } from 'react'
import styled from 'styled-components'
import { space } from 'styled-system'
import { color } from './variants'
import Typography from '../Typography'

const StyledUserAvatar = styled.div`
  ${space}
  display: flex;
  align-items: center;
`

const InitialsWrapper = styled.div`
  ${color}
  width: 40px;
  padding: 10px 0;
  text-align: center;
  background: ${({ theme, color }) => `${theme.colors[color]}`};
  border-radius: 50%;
  margin-right: 10px;
`

const UserAvatar = ({ children, userName, color, displayName, nameColor, ...rest }) => {
  const [nameInitials, setNameInitials] = useState()

  useEffect(() => {
    handleInitials(userName)
  }, [])

  const handleInitials = (name) => {
    let initials = name.match(/\b\w/g) || []
    initials = initials.length > 1 ? ((initials.shift() || '') + (initials.pop() || '')).toUpperCase() : name.substring(0, 2).toUpperCase()
    setNameInitials(initials)
  }

  return (
    <StyledUserAvatar {...rest}>
      <InitialsWrapper color={color} m={0}>
        {nameInitials}
      </InitialsWrapper>
      {!!displayName && (
        <Typography color={nameColor} m={0}>
          {displayName}
        </Typography>
      )}
    </StyledUserAvatar>
  )
}

UserAvatar.defaultProps = {
  color: 'black',
  nameColor: 'black',
}

export default UserAvatar
