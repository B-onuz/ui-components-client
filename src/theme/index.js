const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
