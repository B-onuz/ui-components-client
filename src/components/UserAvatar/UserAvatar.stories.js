import React from 'react'
import UserAvatar from './UserAvatar'
import { Box, Flex } from 'reflexbox'

export default {
  title: 'UserAvatar',
  component: UserAvatar,
}

export const UserAvatarExample = () => <UserAvatar userName="John" />

export const UserAvatarWithName = () => (
  <Flex flexDirection={'column'}>
    <UserAvatar userName="John The Doe" color={'primary'} displayName="Client" m={2} />
    <UserAvatar userName="John Doe The" displayName="Lawyer" nameColor={'primary'} m={2} />
    <UserAvatar userName="Example Lawyer" displayName="Lawyer" m={2} />
    <UserAvatar userName="Client" displayName="Client" color={'primary'} nameColor={'primary'} m={2} />
  </Flex>
)
