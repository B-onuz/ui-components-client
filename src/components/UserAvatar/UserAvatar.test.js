import React from 'react'
import UserAvatar from './UserAvatar.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('Should render UserAvatar', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <UserAvatar />
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
