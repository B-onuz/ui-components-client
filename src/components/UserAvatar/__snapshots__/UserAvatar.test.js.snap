// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render UserAvatar 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  color: #fff;
  width: 40px;
  padding: 10px 0;
  text-align: center;
  background: #1b1c1d;
  border-radius: 50%;
  margin-right: 10px;
}

<div
  className="c0"
>
  <div
    className="c1"
    color="black"
  />
</div>
`;
